client/valor/coretypes.py

```diff
@@ -1032,13 +1033,120 @@ def evaluate_segmentation(
+            A dictionary of parameters for the metrics used in the evaluation. The keys should be the metrics and the values should be dictionaries of parameters for those metrics.
+
+        Returns
+        -------
+        Evaluation
+            A job object that can be used to track the status of the job and get the metrics of it upon completion.
+        """
+        if not datasets and not filters:
+            raise ValueError(
+                "Evaluation requires the definition of either datasets, dataset filters or both."
+            )
+        if metrics_to_return and not set(metrics_to_return).issubset(
+            MetricType.text_generation()
+        ):
+            raise ValueError(
+                f"The following metrics are not supported for text generation: '{set(metrics_to_return) - MetricType.text_generation()}'"
+            )
+
+        # If no api_key is provided, check the environment variables for an api key.
+        if llm_api_params is not None:
+            if "api_key" not in llm_api_params:
+                if "client" not in llm_api_params:
+                    raise ValueError(
+                        "The client must be specified in the llm_api_params."
+                    )
+
+                if llm_api_params["client"] == "openai":
+                    api_key = os.getenv("OPENAI_API_KEY", None)
+                elif llm_api_params["client"] == "mistral":
+                    api_key = os.getenv("MISTRAL_API_KEY", None)
+                elif llm_api_params["client"] == "mock":
+                    api_key = ""
+                else:
+                    raise ValueError(
+                        "The client specified in llm_api_params is not supported."
+                    )
+
+                if api_key is not None:
+                    llm_api_params["api_key"] = api_key
+
+        bleu_weights = None
+        rouge_types = None
+        rouge_use_stemmer = None
+        if metric_params is not None:
+            if not all(
+                metric in metrics_to_return for metric in metric_params.keys()
+            ):
+                raise ValueError(
+                    "All metrics in metric_params must be in metrics_to_return."
+                )
+
+            if MetricType.BLEU in metric_params:
+                bleu_weights = metric_params[MetricType.BLEU].get("weights")
+
+            if MetricType.ROUGE in metric_params:
+                rouge_types = metric_params[MetricType.ROUGE].get(
+                    "rouge_types"
+                )
+                rouge_use_stemmer = metric_params[MetricType.ROUGE].get(
+                    "use_stemmer"
+                )
+
+        # format request
+        datasets = datasets if isinstance(datasets, list) else [datasets]
+        filters = filters if filters else Filter()
+        request = EvaluationRequest(
+            dataset_names=[dataset.name for dataset in datasets],  # type: ignore - issue #604
```

Comment:
I just left it that way because it is that way for the other 3 task types. Issue 604 isn't specific to text generation, so if this should be removed here, then it should be removed everywhere. 

Looks like #604 is @czaloom ?